agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -720,11 +746,13 @@ func (dg *dockerGoClient) RemoveContainer(ctx context.Context, dockerID string,
 		return resp
 	case <-ctx.Done():
 		err := ctx.Err()
-		// Context has either expired or canceled. If it has timed out,
-		// send back the DockerTimeoutError
 		if err == context.DeadlineExceeded {
+			// send back the DockerTimeoutError
+			// Context has either expired or canceled. If it has timed out,
```

Comment:
nit: the comment statement doesn't make sense.  I think it's missing
`// send back the DockerTimeoutError`